api/envoy/api/v2/cds.proto

```diff
@@ -278,6 +278,16 @@ message Cluster {
   // this setting is ignored.
   google.protobuf.Duration dns_refresh_rate = 16 [(validate.rules).duration.gt = {}];
 
+  // If the DNS failure refresh rate is specified and the cluster type is either
+  // :ref:`STRICT_DNS<envoy_api_enum_value_Cluster.DiscoveryType.STRICT_DNS>`,
+  // or :ref:`LOGICAL_DNS<envoy_api_enum_value_Cluster.DiscoveryType.LOGICAL_DNS>`,
+  // this value is used as the cluster’s DNS refresh rate when requests are failing. If this setting
+  // is not specified, the value defaults to the DNS refresh rate. For cluster types other than
+  // :ref:`STRICT_DNS<envoy_api_enum_value_Cluster.DiscoveryType.STRICT_DNS>`
+  // and :ref:`LOGICAL_DNS<envoy_api_enum_value_Cluster.DiscoveryType.LOGICAL_DNS>`
+  // this setting is ignored.
+  google.protobuf.Duration dns_failure_refresh_rate = 42 [(validate.rules).duration.gt = {}];
```

Comment:
This seems fine from an API review perspective. I'm a little late to this review though, I'm curious if we considered doing dynamic/adaptive back-off? There are other places we do this in Envoy today, e.g. see where `JitteredBackOffStrategy` is used.

Reply:
+1 that would probably be optimal, though would require a more complicated implementation including a cap on the back-off.

Reply:
Addressed in 40ca4e2.